molecule/create.yml

```diff
@@ -0,0 +1,166 @@
+        recreate: false
+        log_driver: json-file
+        command: "{{ item.command | default('bash -c \"while true; do sleep 10000; done\"') }}"
+        privileged: "{{ item.privileged | default(omit) }}"
+        security_opts: "{{ item.security_opts | default(omit) }}"
+        volumes: "{{ item.volumes | default(omit) }}"
+        tmpfs: "{{ item.tmpfs | default(omit) }}"
+        capabilities: "{{ item.capabilities | default(omit) }}"
+        exposed_ports: "{{ item.exposed_ports | default(omit) }}"
+        published_ports: "{{ item.published_ports | default(omit) }}"
+        ulimits: "{{ item.ulimits | default(omit) }}"
+        networks: "{{ item.networks | default(omit) }}"
+        network_mode: "{{ item.network_mode | default(omit) }}"
+        dns_servers: "{{ item.dns_servers | default(omit) }}"
+        env: "{{ item.env | default(omit) }}"
+        restart_policy: "{{ item.restart_policy | default(omit) }}"
+        restart_retries: "{{ item.restart_retries | default(omit) }}"
+      register: dnsserver
+      loop: "{{ molecule_yml.platforms | selectattr('name', 'eq', 'ns') | list }}"
+      async: 7200
+      poll: 0
+
+    - name: Wait for DNS creation to complete
+      async_status:
+        jid: "{{ item.ansible_job_id }}"
+      register: dnsdocker_jobs
+      until: dnsdocker_jobs.finished
+      retries: 300
+      loop: "{{ dnsserver.results }}"
+
+    - name: Add DNS container to inventory
+      add_host:
+        name: ns
+        ansible_connection: docker
+      changed_when: false
+
+    - name: Recover internal_dns_ip (from ns container)
+      set_fact:
+        internal_dns_ip: "{{ dnsdocker_jobs.results[0].ansible_facts.docker_container.NetworkSettings.Networks.external.IPAddress }}"
+
+    - name: Create molecule instance(s)
+      docker_container:
+        name: "{{ item.name }}"
+        docker_host: "{{ item.docker_host | default(lookup('env', 'DOCKER_HOST') or 'unix://var/run/docker.sock') }}"
+        hostname: "{{ item.hostname | default(item.name) }}"
+        image: "{{ item.pre_build_image | default(false) | ternary('', 'molecule_local/') }}{{ item.image }}"
+        state: started
+        recreate: false
+        log_driver: json-file
+        command: "{{ item.command | default('bash -c \"while true; do sleep 10000; done\"') }}"
+        privileged: "{{ item.privileged | default(omit) }}"
+        security_opts: "{{ item.security_opts | default(omit) }}"
+        volumes: "{{ item.volumes | default(omit) }}"
+        tmpfs: "{{ item.tmpfs | default(omit) }}"
+        capabilities: "{{ item.capabilities | default(omit) }}"
+        exposed_ports: "{{ item.exposed_ports | default(omit) }}"
+        published_ports: "{{ item.published_ports | default(omit) }}"
+        ulimits: "{{ item.ulimits | default(omit) }}"
+        networks: "{{ item.networks | default(omit) }}"
+        network_mode: "{{ item.network_mode | default(omit) }}"
+        dns_servers: "{{ internal_dns_ip }}"
```

Comment:
To give weight to my 2 remarks regarding this excessive templating : you really have to look for this in order to find it, it's totally hidden in the middle of these useless options. (And I knew what I was looking for)

Reply:
If I remove all those items from default molecule `create.yml` it'll basically break the molecule abstraction. According to the molecule documentation these variables could be defined in the `molecule.yml` file. Hardcoding variable means molecule will not behave as intended and could be weird. 

But I completely see your point of readability. It should at least be highlighted with a comment or something.